### PR TITLE
Add TOF hits into tracking

### PR DESCRIPTION
**Pull request by yezhenyu2003**
Add TOF as a TrackerSubAssembly_4 into tracker.xml

### CODE REVIEW

**Comment by sly2j** [comment]
@yezhenyu2003 to add a "null" tracker needed to disable the negative endocarp, you can create the following type of empty detector, and add it to the assembly:
https://eicweb.phy.anl.gov/EIC/detectors/athena/-/blob/master/compact/tracking_config_canyonlands.xml#L78
(replace `athena` with `epic`).

**Comment by yezhenyu2003** [comment]
> @yezhenyu2003 to add a "null" tracker needed to disable the negative endcap, you can create the following type of empty detector, and add it to the assembly:
> https://eicweb.phy.anl.gov/EIC/detectors/athena/-/blob/master/compact/tracking_config_canyonlands.xml#L78
> (replace `athena` with `epic`).

Added a null tracker (BackwardTOF) into tof_endcap.xml, but tracker_geometry check is failing

D2A_V:Endcap   ERROR     Misconfiguration in volume building detected.

**Comment by yezhenyu2003** [comment]
Added a null tracker (BackwardTOF) into tof_endcap.xml, but tracker_geometry check is failing

D2A_V:Endcap ERROR Misconfiguration in volume building detected.

**Review by wdconinc — COMMENTED** [review]
Can you address why `BarrelTOF_radius_design = 63.0 cm` becomes `BarrelTOF_radius = 63.0 cm` becomes `BarrelTOF_rmin = 63.0 cm` but then (with `BarrelTOF_rOffset = 16.5*mm`)
```xml
        <barrel_envelope
          inner_r="BarrelTOF_rmin-BarrelTOF_rOffset"
          outer_r="BarrelTOF_rmax+BarrelTOF_rOffset"
```
where the envelope is now outside the 63.0 cm to 66.0 cm range in the detector geometry database, https://eic.jlab.org/Geometry/Detector/Detector-20220929172703.html.

**Comment by yezhenyu2003** [comment]
> Can you address why `BarrelTOF_radius_design = 63.0 cm` becomes `BarrelTOF_radius = 63.0 cm` becomes `BarrelTOF_rmin = 63.0 cm` but then (with `BarrelTOF_rOffset = 16.5*mm`)
> 
> ```
>         <barrel_envelope
>           inner_r="BarrelTOF_rmin-BarrelTOF_rOffset"
>           outer_r="BarrelTOF_rmax+BarrelTOF_rOffset"
> ```
> 
> where the envelope is now outside the 63.0 cm to 66.0 cm range in the detector geometry database, https://eic.jlab.org/Geometry/Detector/Detector-20220929172703.html.

Fixed in tof_barrrel.xml

**Comment by osbornjd** [comment]
As discussed today, this should be ready to go

**Comment by yezhenyu2003** [comment]
> As discussed today, this should be ready to go

Yes, waiting for review/approval

File: compact/tof_barrel.xml
Comment (wdconinc) [inline compact/tof_barrel.xml:118]: You cannot go less than `rmin` at all, not even a millimeter. That will(!) result in conflicts with other volumes.

File: configurations/full.yml
Comment (wdconinc) [inline configurations/full.yml:6]: No commented out parts, please. Just remove.

File: compact/tracker.xml
Comment (wdconinc) [inline compact/tracker.xml:140]: This has overlap with #191 where it is reported as failing by Nicolas. @sly2j was going to look into that, and should review here.

File: compact/tof_endcap.xml
Comment (wdconinc) [inline compact/tof_endcap.xml:176]: Do not leave large blocks of commented geometry in the files. Just remove. (git has a history for that reason)

File: compact/tof_barrel.xml
Comment (yezhenyu2003) [reply compact/tof_barrel.xml]: Fixed

File: configurations/full.yml
Comment (yezhenyu2003) [reply configurations/full.yml]: Fixed

File: compact/tof_endcap.xml
Comment (yezhenyu2003) [reply compact/tof_endcap.xml]: Fixed

File: compact/tracker.xml
Comment (yezhenyu2003) [reply compact/tracker.xml:140]: Yes. The recommendation from Tuesday's CompSW meeting by Joe and Sylvester is to move forward with merging this into main branch, while investigating into #191.

**Review by wdconinc — DISMISSED** [review]

**Comment by wdconinc** [comment]
This should merge as soon as checks complete.

**Comment by wdconinc** [comment]
Cannot auto-update because of conflicts.

**Comment by yezhenyu2003** [comment]
What conflicts?

**Comment by wdconinc** [comment]
That's an automated message. I'd have to look...

**Comment by wdconinc** [comment]
There is a lot of stuff going into the main branch and many things are getting updated, sometimes in the same areas of the code. Take a look at the pull request and it will show you.

**Comment by yezhenyu2003** [comment]
I updated the files to resolve the conflicts. The remaining differences are intentional for TOF.

Zhenyu

**Review by wdconinc — APPROVED** [review]

File: compact/definitions.xml
Comment (wdconinc) [inline compact/definitions.xml:446]: Related to PR?

File: compact/definitions.xml
Comment (wdconinc) [inline compact/definitions.xml:506]: Related to PR?

File: compact/definitions.xml
Comment (wdconinc) [inline compact/definitions.xml:539]: Related to PR?

File: compact/definitions.xml
Comment (wdconinc) [inline compact/definitions.xml:549]: Related to PR?

**Review by wdconinc — CHANGES_REQUESTED** [review]

File: compact/definitions.xml
Comment (yezhenyu2003) [reply compact/definitions.xml]: fixed

File: compact/definitions.xml
Comment (yezhenyu2003) [reply compact/definitions.xml]: fixed

File: compact/definitions.xml
Comment (yezhenyu2003) [reply compact/definitions.xml]: fixed

File: compact/definitions.xml
Comment (yezhenyu2003) [reply compact/definitions.xml]: fixed

**Review by wdconinc — APPROVED** [review]

**Pull request merged**
